### PR TITLE
Implement image based lighting

### PR DESCRIPTION
**Pull request by JaimeBea**
This is a diffuse and specular IBL implementation using HDR images. Skyboxes have been changed to use HDR images directly. These images get transformed into cubemaps inside the engine.

The current resolutions used for the different cubemaps are as follows:
- Environment cubemap (512x512 with mips)
- Irradiance cubemap (128x128)
- Pre-Filtered cubemap (128x128 with mips)
- Environment BRDF image (512x512)

image

### CODE REVIEW

**Review by magalenyo — APPROVED** [review]

**Review by Jordi-Gil — APPROVED** [review]

File: Project/Source/Resources/ResourceSkybox.cpp [hunk 1]
```diff
@@ -100,61 +72,167 @@ void ResourceSkybox::Load() {
...
 
 	unsigned timeMs = timer.Stop();
 	LOG("Skybox loaded in %ums.", timeMs);
 }
 
 void ResourceSkybox::Unload() {
 	if (glCubeMap) glDeleteTextures(1, &glCubeMap);
-	if (vao) glDeleteVertexArrays(1, &vao);
-	if (vbo) glDeleteBuffers(1, &vbo);
+	if (glIrradianceMap) glDeleteTextures(1, &glIrradianceMap);
```
Comment (lowysole) [inline Project/Source/Resources/ResourceSkybox.cpp:237]: Isn't necessary to delete the other textures?

File: Game/Assets/Shaders/imageBasedLighting.shader [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+		vec3 L = reflect(-V, H);
+		float NL = max(dot(N, L), 0.0);
+		float NH = max(dot(N, H), 0.0);
+		float VH = max(dot(V, H), 0.0);
+		if (NL > 0) {
+			float D = GGXNormalDistribution(NH, roughness);
+			float pdf = (D * NH / (4.0 * VH)) + 0.0001;
+			float saTexel  = 4.0 * PI / (6.0 * environmentResolution * environmentResolution);
+			float saSample = 1.0 / (float(NUM_SAMPLES) * pdf + 0.0001);
+			float mipLevel = roughness == 0.0 ? 0.0 : 0.5 * log2(saSample / saTexel);
```
Comment (Mefiso) [inline Game/Assets/Shaders/imageBasedLighting.shader:141]: Why do you have these lines here? I don't think you need them to compute pre-filtered environment map, and you are not using them here.

**Review by Mefiso — DISMISSED** [review]

File: Game/Assets/Shaders/imageBasedLighting.shader (hunk 2 above)
Comment (JaimeBea) [reply Game/Assets/Shaders/imageBasedLighting.shader:141]: Whoops, I forgot to add mipLevel to the next line. These lines are there to reduce the artifacts on reflections for higher roughness levels.

**Review by FrancPS — APPROVED** [review]
As a PR mostly about Shaders and Render, if visualisation is correct and Game works, you can merge

**Review by lowysole — APPROVED** [review]

**Review by Mefiso — APPROVED** [review]

**Pull request merged**